docs/book/src/developer/tilt.md

```diff
@@ -222,21 +222,23 @@ kustomize_substitutions:
 Set to `manual` to disable auto-rebuilding and require users to trigger rebuilds of individual changed components through the UI.
 
 **extra_args** (Object, default={}): A mapping of provider to additional arguments to pass to the main binary configured
-for this provider. Each item in the array will be passed in to the manager for the given provider.
+for this provider. Each item in the array will be passed in to the manager for the given provider. (Note: Some args are already
```

Comment:
Nope I think you're right :)

Reply:
I think we could possibly fix that by appending the extra_args to cmd instead. 
I would have to take a closer look how that currently works, but I think it should be possible.

Reply:
Yep, that sounds like a solution. I think it is called `container_args` for [Tilt](https://docs.tilt.dev/api.html#api.docker_build), which take precedence over args specified in k8s yaml.

Reply:
Hm that's a good point. I think they shouldn't be overwritten. 

I'll have to take some time and take a closer look at our current code.

Reply:
Sure, thx. Another, more complicated, way might be to do this in tilt-prepare, like @fabriziopandini said in the issue. Probably like doing it with debug and observability and pass in the args to tilt-prepare and collect the args from the deployment and only overwrite them if those are explicity set in extra_args.

Reply:
I would prefer to have a single point to manipulate manifests including handling extra_args.
I also think we should make tilt prepare to read tilt_settings at this point

Reply:
Took a closer look and I agree with Fabrizio.
I think we're now at a point where it would be good to move the more complicated logic to tilt-prepare. 
This should be mostly (and maybe only) related to calculating command and entrypoint. (but I would definitely only move those parts in the next iteration)

It roughly comes down to:
* Drop the logic for entrypoint calculation in Tiltfile: ~ https://github.com/kubernetes-sigs/cluster-api/blob/8cf203667e7aa08e72e841f8992b422c7cad7adf/Tiltfile#L181-L258
* Implement parsing the tilt-settings.{yaml|json} in tilt-prepare/main.go
    * Let's use a struct for that to make it more type-safe. (TBD if we want to add all fields or only the ones actually used in tilt-prepare)
* Implement the entrypoint/command calculation in tilt-prepare/main.go (with the improved semantic we want)

@johannesfrey Would you be interested in tackling this? (I can provide further details/guidance if needed)

Reply:
Sure, I will take a shot and will contact you if I need more details/help :).

Reply:
@johannesfrey thanks for taking a look, really appreciated it!
I'm aware that the scope is very different from the original PR, so I will be happy to help and have a deep dive session together; I'm pretty sure also @sbueringer will be happy to join 😉. 
Also, don't hesitate to speak up if you feel this is becoming overwhelming or simply challenging given the time you can allocate to this task; I think there is room to break down the work into smaller items and eventually federate the work with someone else.

Reply:
Thanks, that sounds good. I probably won't come very far today but, if that's ok with you (@fabriziopandini, @sbueringer) I will ping you beginning next week in Slack!?

Reply:
No rush, sounds good!

Reply:
I pushed a first rough idea to: https://github.com/kubernetes-sigs/cluster-api/pull/6214
If ok, this might serve as a first basis for a deep dive session. I already put some questions as comments to the PR.

Would it make sense to close this PR in favor of https://github.com/kubernetes-sigs/cluster-api/pull/6214 and continue there?

Reply:
@johannesfrey Sounds good to me! (closing the PR and continuing on the other one)